### ISSUE TITLE
Generated java client does not parse java 8 dates

### ISSUE THREAD

**Issue by jonocodes** [swagger-api/swagger-codegen#2386]
I have a client that uses java.time.LocalDate

The problem is when the client reads a local date (ie - '2000-01-01'), it throws an error:

Caused by: com.sun.jersey.api.client.ClientHandlerException: com.fasterxml.jackson.databind.JsonMappingException: Can not instantiate value of type [simple type, class java.time.LocalDate] from String value ('2000-01-01'); no single-String constructor/factory method
at [Source: java.io.BufferedInputStream@2cc7a2a9; line: 1, column: 588] (through reference chain: org.plos.ned_client.model.IndividualComposite["groups"]->java.util.ArrayList[0]->org.plos.ned_client.model.Group["startdate"])

My current workaround is to do handle my own serialization like this:
http://stackoverflow.com/questions/27952472/serialize-deserialize-java-8-java-time-with-jackson-json-mapper

But I would think use of LocalDate would also allow JDK8 parsing to work without needing special handling.

**Comment by jonocodes** [comment]
I have tried using this both in my client and in the consumer of my client:
https://github.com/FasterXML/jackson-datatype-jdk8

to no avail. I will take a look at the sample here:
https://github.com/fehguy/swagger-codegen/tree/sample/java8/samples/client/petstore/java/default

**Comment by jonocodes** [comment]
Trying to implement the above sample in my project to no avail.

java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.mojo.exec.ExecJavaMojo$1.run(ExecJavaMojo.java:297)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalAccessError: tried to access method com.fasterxml.jackson.databind.ser.std.StdSerializer.<init>(Ljava/lang/Class;)V from class com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
	at com.fasterxml.jackson.datatype.jsr310.JavaTimeModule.<init>(JavaTimeModule.java:159)
	at org.plos.ned_client.ApiClient.<init>(ApiClient.java:87)
	at org.plos.ned.App.main(App.java:19)

Will come back to this when I have more time.

**Comment by ePaul** [comment]
In order to judge whether this is a bug on our side or a mistake on your side (and if so, how to solve it), could you post your API specification and the codegen settings you have used (e.g. command line or maven plugin configuration)?

**Comment by jonocodes** [comment]
Ah. Ok I got it working. Thanks. I think I was importing a conflicting version of jackson-datatype-jdk8. 

Turns out my client consumers are not up to jdk 8, so I need to support JodaTime, which I should be able to do.

Regardless, I opened this ticket to help promote the support of Java 8. If this flag is specified
"dateLibrary": "java8"

the ApiClient should automatically support the Jdk8Module. Without manual updates to the generated code.
